mlrun/projects/project.py

```diff
@@ -2180,16 +2180,13 @@ def _instantiate_function(
         if func is None and not _has_module(handler, kind):
             # if function path is not provided and it is not a module (no ".")
             # use the current notebook as default
-            if not is_ipython:
-                raise ValueError(
-                    "Function path or module must be specified (when not running inside a Notebook)"
-                )
-            from IPython import get_ipython
+            if is_ipython:
+                from IPython import get_ipython
```

Comment:
Before this change if this was not `is_ipython` this method would explode and now it won't. Is this intentional?

Reply:
Yes because I wanted to allow creating nuclio functions without code (application runtime is created with an image instead). If no code is given and it's not a nuclio runtime, it will explode in `_init_function_from_dict`